app/models/server/models/Rooms.js

```diff
@@ -544,24 +544,53 @@ export class Rooms extends Base {
 		return this._db.find(query, options);
 	}
 
-	findByNameOrFNameAndTypeIncludingTeamRooms(name, type, teamIds, options) {
+	findByNameOrFNameAndRoomIdsIncludingTeamRooms(name, teamIds, roomIds, options) {
 		const query = {
-			t: type,
+			t: {
+				$in: [
+					'p',
+					'c',
+				],
+			},
 			teamMain: {
 				$exists: false,
 			},
 			$and: [
 				{
 					$or: [
 						{
-							teamId: {
-								$exists: false,
-							},
+							$and: [
```

Comment:
You can remove the `$and`s in pretty much all the clauses here.

Reply:
Great, I've removed all the additional `$and`s (a lot easier to read the query now 😆 )